src/etc/inc/dyndns.class

```diff
@@ -590,6 +602,87 @@
 					curl_setopt($ch, CURLOPT_URL, $server);
 					curl_setopt($ch, CURLOPT_POSTFIELDS, $post_data);
 					break;
+				case 'hetzner':
+				case 'hetzner-v6':
+					// API documentation: https://dns.hetzner.com/api-docs
+					$server = 'https://dns.hetzner.com/api/v1';
+					$fqdn = str_replace(' ', '', $this->_dnsHost);
+					$recordType = $this->_useIPv6 ? "AAAA" : "A";
+					$ttlData = intval($this->_dnsTTL) < 1 ? 120 : intval($this->_dnsTTL);
+					$hostData = [
+						"value" => "{$this->_dnsIP}",
+						"type" => $recordType,
+						"name" => "",
+						"ttl" => $ttlData,
+						"zone_id" => ""
+					];	
+
+					$headerAuth = [
+						"Auth-API-Token: {$this->_dnsPass}",
+						'Content-Type: application/json'
+					];
+		
+					curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+					curl_setopt($ch, CURLOPT_HTTPHEADER, $headerAuth);
+
+					// Get all zone info
+					$zonesUrl = "$server/zones";
+					curl_setopt($ch, CURLOPT_URL, $zonesUrl);
+					$output = json_decode(curl_exec($ch));
+					$http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+					$zoneId = null; // Set default value
+
+					if (!empty($output->zones)) {
+						// Iterate zone objects, check if $fqdn is equal to or ends with zone name
+						foreach ($output->zones as $key => $zoneObj) {
+							if (preg_match("/^{$zoneObj->name}$|\.{$zoneObj->name}$/", $fqdn)) {
+								// Found matching zone
+								$zoneId = $zoneObj->id;
+								// Get $hostName from $fqdn, set $domainName
+								// These are only really used for log messages.
+								$hostName = preg_replace("/\.?{$zoneObj->name}$/", '', $fqdn);
+								$domainName = $zoneObj->name;
+								break;
+							}
+						}
+					}
+							
+					if ($zoneId) { // If zone ID was found get host ID
+						$dnsRecordsUrl = "$server/records?zone_id=$zoneId";
```

Comment:
Similar to last comment, surround variables with braces inside the string.

Reply:
done